subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publication/DefaultIvyPublication.java

```diff
@@ -94,6 +103,8 @@ public int compare(UsageContext left, UsageContext right) {
             return left.getUsage().getName().compareTo(right.getUsage().getName());
         }
     };
+    @VisibleForTesting
+    public static final String INCOMPATIBLE_FEATURE = " uses dependency management feature(s) that will most likely make the produced Ivy file incomplete and/or not consumable by Ivy compatible solutions";
```

Comment:
This message is a bit wordy for my liking:

"contains dependencies that cannot be represented in a published [pom file | ivy descriptor]."
"contains dependencies that will produce a pom file that cannot be consumed by a Maven client"
"contains dependencies that will produce an Ivy descriptor that cannot be consumed by an Ivy client"

I'm not sure that there's a real use case for the last one.

Reply:
Reworked based on your suggestion